web/packages/teleport/src/services/auth/auth.ts

```diff
@@ -149,20 +162,43 @@ const auth = {
 
         return api.put(cfg.getPasswordTokenUrl(), request);
       })
-      .then(makeChangedUserAuthn);
+      .then(j => {
+        if (eventMeta) {
+          userEventService.capturePreUserEvent({
+            event: CaptureEvent.PreUserOnboardSetCredentialSubmitEvent,
+            username: eventMeta.username,
+          });
+
+          userEventService.capturePreUserEvent({
+            event: CaptureEvent.PreUserOnboardRegisterChallengeSubmitEvent,
+            username: eventMeta.username,
+            mfaType: eventMeta.mfaType,
+            loginFlow: deviceUsage,
+          });
```

Comment:
Isn't this also potentially subject to reordering due to random network hiccups?

Reply:
Yep, but, handling what we can control. Lets see how much that manifests